xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -923,20 +920,18 @@ void CPVRClients::ProcessMenuHooks(int iClientID, PVR_MENUHOOK_CAT cat, const CF
       pDialog->Reset();
       pDialog->SetHeading(CVariant{19196});
 
-      PVR_CLIENTMAP_CITR itrClients;
-      for (itrClients = clients.begin(); itrClients != clients.end(); itrClients++)
+      for (const auto client : clients)
       {
-        pDialog->Add(itrClients->second->GetBackendName());
+        pDialog->Add(client.second->GetBackendName());
       }
       pDialog->Open();
 
       int selection = pDialog->GetSelectedItem();
       if (selection >= 0)
       {
-        itrClients = clients.begin();
-        for (int i = 0; i < selection; i++)
-          itrClients++;
-        iClientID = itrClients->first;
+        auto client = clients.begin();
+        std::advance(client, selection);
+        iClientID = client->first;
```

Comment:
Hi @ksooo and @fritsch have changed it and learned a new part from C++, thanks a lot! :D